packages/xrpl/test/integration/submitAndWait.test.ts

```diff
@@ -0,0 +1,138 @@
+/* eslint-disable @typescript-eslint/no-misused-promises -- supposed to return a promise here */
+/* eslint-disable no-restricted-syntax -- not sure why this rule is here, definitely not needed here */
+import { assert } from 'chai'
+import { AccountSet, convertStringToHex, ValidationError } from 'xrpl-local'
+
+import { assertRejects } from '../testUtils'
+
+import serverUrl from './serverUrl'
+import {
+  setupClient,
+  teardownClient,
+  type XrplIntegrationTestContext,
+} from './setup'
+import { ledgerAccept } from './utils'
+
+// how long before each test case times out
+const TIMEOUT = 60000
+
+describe('client.submitAndWait', () => {
+  let testContext: XrplIntegrationTestContext
+
+  beforeEach(async () => {
+    testContext = await setupClient(serverUrl)
+  })
+  afterEach(async () => teardownClient(testContext))
+
+  it(
+    'submitAndWait an unsigned transaction',
+    async () => {
+      const accountSet: AccountSet = {
+        TransactionType: 'AccountSet',
+        Account: testContext.wallet.classicAddress,
+        Domain: convertStringToHex('example.com'),
+      }
+
+      let retries = 10
+
+      while (retries > 0) {
+        retries -= 1
+        const responsePromise = testContext.client.submitAndWait(accountSet, {
+          wallet: testContext.wallet,
+        })
+        const ledgerPromise = setTimeout(ledgerAccept, 1000, testContext.client)
+
+        try {
+          // eslint-disable-next-line no-await-in-loop -- Testing purposes
+          const [response, _ledger] = await Promise.all([
+            responsePromise,
+            ledgerPromise,
+          ])
+
+          assert.equal(response.type, 'response')
+          assert.equal(response.result.validated, true)
+          retries = 0
```

Comment:
Does `break` not work here because it's in a try/catch?

Reply:
Break should work